### DISCUSSION TITLE
Pass Variable via URL

### DISCUSSION THREAD

**Discussion by Sagestum** [Budibase/budibase#10750] [Help] [answered]
Hi, I've read the instructions but I'm not quite sure how to work with URL variables. My understanding is that I should specify a variable in the URL like this:

https://your-budibase-app.com/path-to-page**?parameter1=value1**

But how do I access it in Budibase? I want to send an email to a customer and the URL should contain the ticket number, which is then stored in Budibase. It's like asking "How satisfied were you with the ticket handling?

And my guess is that this could be accessed via the {{URL.ticketNumber}} function.

But nothing is returned :(

**Comment by melohagan** [comment, marked as answer]
Hey @Sagestum 

The Budibase URL variables are path variables and not query variables.

As such they need to be in the format as outlined in the docs:

**Screen route within the Budibase builder: (Note the colon)**
`https://myapp.com/app/screen/:param1`

Reference the URL variable with the binding: `{{ URL.param1 }}`

Then use this URL in the app:
`https://myapp.com/app/screen/<--the actual value for param1 here-->`

**Comment by ahmedelnaqa** [comment]
I follow same steps but the records not changed and shown only the last record

**Reply by ahmedelnaqa** [reply]
I solve it, in one way:
- the filter with binding variable adds to the repeater instead of the data provider.

is it right???

**Reply by melohagan** [reply]
Hey @ahmedelnaqa The repeater block will allow you to filter, and is a combination of:

> Data provider
> > Repeater

The reason this is used is because a Data provider that is filtered on a unique ID will return a singleton array (a collection of one value).
Then the repeater is used to "repeat" over that collection, in this case a single item.
